raiden/tests/utils/blockchain.py

```diff
@@ -38,6 +38,7 @@ def wait_until_block(chain, block):
     curr_block = chain.block_number()
     while curr_block < block:
         curr_block = chain.next_block()
+        gevent.sleep(0)
```

Comment:
haha ... I used to have this here but removed it when we removed tester the last time. Are we sure that the `0` value allows for a context switch?

Edit: According to the [docs](http://www.gevent.org/api/gevent.html#gevent.sleep)

> In the current implementation, a value of 0 (the default) means to yield execution to any other runnable greenlets, but this greenlet may be scheduled again before the event loop cycles (in an extreme case, a greenlet that repeatedly sleeps with 0 can prevent greenlets that are ready to do I/O from being scheduled for some (small) period of time); a value greater than 0, on the other hand, will delay running this greenlet until the next iteration of the loop.

So perhaps just switch to a very small positive value instead?

Reply:
yep, context switch. I was getting infinite loop with tester in there and this seems to help. Good point referring to the docs, I can put something like 1 ms as delay.